src/main/java/org/springframework/guice/annotation/ModuleRegistryConfiguration.java

```diff
@@ -86,12 +92,27 @@ class ModuleRegistryConfiguration implements BeanDefinitionRegistryPostProcessor
 
 	private static final String SPRING_GUICE_STAGE_PROPERTY_NAME = "spring.guice.stage";
 
+	private static final List<String> SPRING_GUICE_IGNORED_ANNOTATION_PREFIXES = Arrays
+			.asList("com.google.inject.multibindings", "com.google.inject.internal.UniqueAnnotations");
```

Comment:
I guess the `multibindings` prefix can be ignored, this logic now handles them correctly and the annotations are in `com.google.inject.internal` on 5.1. 

I didn't look into `UniqueAnnotations` but I wonder if there's also a similar general change that would deal with those correctly too.